### DISCUSSION TITLE
Parameter only for POST

### DISCUSSION THREAD

**Discussion by DenisFerrero** [OAI/OpenAPI-Specification#2865] [Q&A] [answered]
I have a schema that I would like to use for both POST and PUT methods but a target parameter is allowed only for the POST request.

To make it clear let's imagine that there is a possibility to **create** and **update** a *user*.
At the POST it's required the name and the inviteCode, then for the PUT just the name and don't allow to edit the inviteCode anymore

```yaml
components:
  schemas:
    user:
      type: object
      properties:
        name:
          type: string
        inviteCode:
          type: string
paths:
  "/user":
    post:
      requestBody:
        content:
          application/json:
            schema:
              "$ref": "#/components/schemas/user"
  "/user/{id}":
    put:
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: string
      requestBody:
        content:
          application/json:
            schema:
              "$ref": "#/components/schemas/user"
```

Is there a way to do it by just adding a parameter to the ```inviteCode```?

**Comment by webron** [comment, marked as answer]
JSON Schema doesn't quite allow you to pick and choose which properties you want to use from a given schema definition.
You have two options:
- split `user` into two schemas, where one uses `allOf` the other and adds the `inviteCode`
- use the `user` schema for both, but in the case of put make it a combination of `allOf` and `not` to remove support for `inviteCode`

**Reply by DenisFerrero** [reply]
Ok thanks a lot

**Comment by handrews** [comment]
Answer accepted by OP, closing.
